ethcore/wasm/src/runtime.rs

```diff
@@ -161,23 +161,17 @@ impl<'a> PendingTransaction for Runtime<'a> {
 		}
 
 		match call_result {
-			MessageCallResult::Success(gas_left, return_data)
-			| MessageCallResult::Reverted(gas_left, return_data) => {
-				// cannot overflow, before making call gas_counter was incremented with gas, and gas_left < gas
-				let gas_used = gas_left.low_u64() * self.ext.schedule().wasm().opcodes_div as u64
-					/ self.ext.schedule().wasm().opcodes_mul as u64;
-				let gas_left = match self.gas_counter.checked_sub(gas_used) {
-					Some(gas_left) => gas_left,
-					None => {
-						receipt.outcome = TransactionOutcome::InsufficientGas;
-						receipt.gas_used = self.gas_limit;
-						return receipt;
-					}
-				};
-				self.gas_counter -= gas_left;
-				receipt.gas_used = gas_used;
+			MessageCallResult::Success(post_gas_left, return_data)
+			| MessageCallResult::Reverted(post_gas_left, return_data) => {
+				receipt.gas_used = pre_gas_left - post_gas_left.low_u64();
+				// ^ by definition pre_gas_left > post_gas_left so this cannot overflow
+				if !self.charge_gas(receipt.gas_used) {
+					// ^ cost for xcc is not be "adjusted" because functions called
+					// during its scope are already individually adjusted.
+					receipt.outcome = TransactionOutcome::InsufficientGas;
```

Comment:
Not for a good reason, but if you're curious, it's because this receipt isn't actually used anywhere. The transaction is either successful and the contents are returned to the caller, or else the transaction is aborted (as it is here) and this "application-layer" receipt isn't returned.